napari/components/_layer_slicer.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import annotations
+
+import logging
+from concurrent.futures import Executor, Future, ThreadPoolExecutor
+from threading import RLock
+from typing import Dict, Iterable, Optional, Tuple
+
+from napari.components import Dims
+from napari.layers import Layer
+from napari.utils.events.event import EmitterGroup, Event
+
+logger = logging.getLogger("napari.components._layer_slicer")
+
+
+class _LayerSlicer:
```

Comment:
The goal here was to avoid accidental public exposure through imports in public modules. For example, if this is `LayerSlicer` and is imported in `viewer_model.py` (as it will be in the future), then someone can write `from napari.components.viewer_model import LayerSlicer`.

I'm not sure what the accepted standard is in Python, but by prepending the `_` to the name of the class of the function, it is always clear that this is a private implementation detail that may change in the future, regardless of where it's imported.

Reply:
the standard for Python is to define `__all__ = ("names", )` with list of thins that could be imported from module. And if class/function is not listed in all then should not be imported.

Reply:
I think it just makes absolutely sure there is no way to confuse this class for a public module.